### ISSUE TITLE
Support webasm

### ISSUE THREAD

**Issue by davidrusu** [spacejam/rust-crdt#16]
In github.com/hermits-grove/mona we have a webextension for firefox. I would like to have copies of the crdt's in the JS world just as I have them in Rust.

**Comment by chpio** [comment]
But there is nothing in `crdts` that isn't supported by wasm? (the blocker is more of a js problem than a wasm problem?)

**Comment by davidrusu** [comment]
@chpio yep very likely true, do you want to experiment with setting up support for wasm?

**Comment by Spoonbender** [comment]
@davidrusu I started experimenting with adding wasm support for rust-crdt, and I've encountered some issues.
How would you like to communicate about this experiment, if at all? Over this GitHub issue, or some other place?

**Comment by davidrusu** [comment]
Github Issues is fine if you've got some concrete issues to work through, otherwise for open-ended conversations lets try the `discussions` tab, I've just enabled this github feature now: https://github.com/rust-crdt/rust-crdt/discussions

**Comment by timfpark** [comment]
I'm not sure if more investigation happened elsewhere, but I just had a high level look at this.

Exposing `crdts` directly is going to be difficult given the current state of the art with `wasm-bindgen` and how `crdts` is designed as you can't use it to expose items with type parameters with `wasm-bindgen` through to JavaScript. 

Given this, [I prototyped a wrapping based approach](https://github.com/timfpark/rust-crdt-wasm) to show how this could be accomplished for folks that want to utilize the same library in a hybrid Rust / JavaScript stack (see the `README.md` for more details).  This approach is probably better suited for users of the library to implement for their specific use cases as you have to materialize the CRDT and actor type combination.  That said, I have only an intermediate level of knowledge of Rust, so would love to be told there is a better way that we could avoid this limitation of `wasm-bindgen` as well.

**Comment by davidrusu** [comment]
@timfpark very cool, I skimmed the code briefly, I'll have more time tomorrow to try it out
